exporters/otlp/src/otlp_grpc_log_exporter.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include "opentelemetry/exporters/otlp/otlp_grpc_log_exporter.h"
+#  include "opentelemetry/exporters/otlp/otlp_log_recordable.h"
+#  include "opentelemetry/exporters/otlp/otlp_recordable_utils.h"
+
+// clang-format off
+
+#  include "opentelemetry/exporters/otlp/protobuf_include_prefix.h"
+
+#  include "opentelemetry/proto/collector/logs/v1/logs_service.pb.h"
+#  include "opentelemetry/proto/collector/logs/v1/logs_service.grpc.pb.h"
+
+#  include "opentelemetry/exporters/otlp/protobuf_include_suffix.h"
+
+// clang-format on
+
+#  include "opentelemetry/ext/http/common/url_parser.h"
+#  include "opentelemetry/sdk/common/global_log_handler.h"
+
+#  include <chrono>
+#  include <fstream>
+#  include <memory>
+
+#  include <grpcpp/grpcpp.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace otlp
+{
+
+// ----------------------------- Helper functions ------------------------------
+// TODO: move exporters shared function to OTLP common library.
+static std::string get_file_contents(const char *fpath)
+{
+  std::ifstream finstream(fpath);
+  std::string contents;
+  contents.assign((std::istreambuf_iterator<char>(finstream)), std::istreambuf_iterator<char>());
+  finstream.close();
+  return contents;
+}
+
+struct OtlpGrpcExporterOptions;
+
+/**
+ * Create gRPC channel from the exporter options.
+ */
+static std::shared_ptr<grpc::Channel> MakeGrpcChannel(const OtlpGrpcExporterOptions &options)
+{
+  std::shared_ptr<grpc::Channel> channel;
+
+  //
+  // Scheme is allowed in OTLP endpoint definition, but is not allowed for creating gRPC channel.
+  // Passing URI with scheme to grpc::CreateChannel could resolve the endpoint to some unexpected
+  // address.
+  //
+
+  ext::http::common::UrlParser url(options.endpoint);
+  if (!url.success_)
+  {
+    OTEL_INTERNAL_LOG_ERROR("[OTLP Exporter] invalid endpoint: " << options.endpoint);
+
+    return nullptr;
+  }
+
+  std::string grpc_target = url.host_ + ":" + std::to_string(static_cast<int>(url.port_));
+
+  if (options.use_ssl_credentials)
+  {
+    grpc::SslCredentialsOptions ssl_opts;
+    if (options.ssl_credentials_cacert_path.empty())
+    {
+      ssl_opts.pem_root_certs = options.ssl_credentials_cacert_as_string;
+    }
+    else
+    {
+      ssl_opts.pem_root_certs = get_file_contents((options.ssl_credentials_cacert_path).c_str());
+    }
+    channel = grpc::CreateChannel(grpc_target, grpc::SslCredentials(ssl_opts));
+  }
+  else
+  {
+    channel = grpc::CreateChannel(grpc_target, grpc::InsecureChannelCredentials());
+  }
+
+  return channel;
```

Comment:
Ok. now I understand the split. Do you think we can move both these helper functions to common place as part of this PR if this is not a lot of effort?

Reply:
Yes, move it to common was the intention, but it moves gRPC dependence there as well. How about we do that in a separate PR?